### PR TITLE
Relax `run_exports` constraint to 'x.x'

### LINKED ISSUE #144: Relax PROJ 9 pin
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

`run_exports` is pinned to `x.x.x`. This has been discussed 5 years ago in https://github.com/conda-forge/proj.4-feedstock/issues/64

I like to reopen this issue because at least since version 8.0.0, backward compatibility and semantic versioning compliance looks very solid: https://abi-laboratory.pro/?view=timeline&l=proj

### Installed packages

```shell
n/a
```

### Environment info

```shell
n/a
```

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

fixes #144

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,11 +11,11 @@ source:
...
 
 build:
-  number: 1
+  number: 2
   run_exports:
-    # so name changes in bugfix revisions.  Pin to bugfix revision.
+    # ABI has been stable across minor versions since 8.0.0
     #    https://abi-laboratory.pro/tracker/timeline/proj/
-    - {{ pin_subpackage('proj', max_pin='x.x.x') }}
+    - {{ pin_subpackage('proj', max_pin='x.x') }}
```
Comment (xylar) [inline recipe/meta.yaml:18]: @speleo3, please check that this is what you were hoping for in #144.

**Comment by xylar** [comment]
I have a branch ready for a PR to conda-forge-pinning to loosen the pin there, too:
https://github.com/xylar/conda-forge-pinning-feedstock/tree/loosen-proj

**Comment by xylar** [comment]
@conda-forge/proj-4, any concerns?  Otherwise, I'll merge this tomorrow (assuming tests pass).

File: recipe/meta.yaml (hunk 1 above)
Comment (speleo3) [reply recipe/meta.yaml:18]: Looks good, thank you!

**Comment by xylar** [comment]
Okay, we're good to go.

**Review by mwtoews — APPROVED** [review]

**Comment by xylar** [comment]
Please see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6006

**Pull request merged**